kv/memberlist/memberlist_client_test.go

```diff
@@ -209,16 +209,11 @@ func getData(t *testing.T, kv *Client, key string) *data {
 	return nil
 }
 
-func cas(t *testing.T, kv *Client, key string, updateFn func(*data) (*data, bool, error)) {
-	t.Helper()
-
-	if err := casWithErr(context.Background(), t, kv, key, updateFn); err != nil {
-		t.Fatal(err)
-	}
+func cas(kv *Client, key string, updateFn func(*data) (*data, bool, error)) error {
```

Comment:
Why the change to return an error instead of failing the test on error (as a `t.Helper()`)? I think normally it's preferable to catch test errors in test helpers instead of returning them.